parquet-tools/src/main/java/parquet/tools/read/SimpleRecord.java

```diff
@@ -97,6 +99,92 @@ public void prettyPrint(PrintWriter out, int depth) {
     }
   }
 
+  public void prettyPrintJson(PrintWriter out) {
+    Boolean isArray = isArray();
+    Boolean isMap = isMap();
+    if (isArray) {
+      out.print("[");
+    } else {
+      out.print("{");
+    }
+    boolean comma = false;
+
+    for (NameValue value : values) {
+      if (comma) {
+        out.print(",");
+      } else {
+        comma = true;
+      }
+
+      if (!isArray && !isMap) {
+        out.print("\"" + value.getName() + "\":");
+      }
+      Object val = value.getValue();
+
+      if (isMap) {
+        SimpleRecord kv = (SimpleRecord) val;
+        out.print("\"" + kv.getValues().get(0).getValue() + "\":");
+        printValueJson(out, kv.getValues().get(1).getValue());
+      } else {
+        printValueJson(out, val);
+      }
+    }
+
+    if (isArray) {
+      out.print("]");
+    } else {
+      out.print("}");
+    }
+  }
+
+  private void printValueJson(PrintWriter out, Object val) {
+    if (val == null) {
+      out.print("null");
+    } else if (byte[].class == val.getClass()) {
+      out.print("\"");
+      out.print(Arrays.toString((byte[])val));
+      out.print("\"");
+    } else if (short[].class == val.getClass()) {
+      out.print(Arrays.toString((short[])val));
+    } else if (int[].class == val.getClass()) {
+      out.print(Arrays.toString((int[])val));
+    } else if (long[].class == val.getClass()) {
+      out.print(Arrays.toString((long[])val));
+    } else if (float[].class == val.getClass()) {
+      out.print(Arrays.toString((float[])val));
+    } else if (double[].class == val.getClass()) {
+      out.print(Arrays.toString((double[])val));
+    } else if (boolean[].class == val.getClass()) {
+      out.print(Arrays.toString((boolean[]) val));
+    } else if (val.getClass().isArray()) {
+      out.print(Arrays.deepToString((Object[])val));
+    } else if (SimpleRecord.class.isAssignableFrom(val.getClass())) {
+      ((SimpleRecord) val).prettyPrintJson(out);
+    } else if (val.getClass() == String.class) {
+      out.print("\"" + String.valueOf(val) + "\"");
```

Comment:
JSON requires certain characters, like newline, to be escaped. This is done automatically with Jackson, which I would recommend using instead of building JSON manually.